test/stream/api/sink/update-006-invalid-jsonnet-template/006-poll-sink/expected-response.json

```diff
@@ -0,0 +1,11 @@
+{
+  "taskId": "api.update.sink/%%TEST_DEFAULT_BRANCH_ID%%/my-source/my-sink/%s",
+  "type": "api.update.sink",
+  "url": "https://stream.keboola.local/v1/tasks/api.update.sink/%%TEST_DEFAULT_BRANCH_ID%%/my-source/my-sink/%s",
+  "status": "error",
+  "isFinished": true,
+  "createdAt": "%s",
+  "finishedAt": "%s",
+  "duration": %d,
+  "error": "column \"jsonnet\" template is invalid: 1:1-8 Unknown variable: Invalid"
```

Comment:
First check, ... there are 2 methods: `StartTask` (in the background), `RunTask` (in the foreground):
- :heavy_check_mark: 
  - Both (all) endpoints use `startTask`:
    - https://github.com/keboola/keboola-as-code/blob/ea25882687e215d2e91c352d4aa9b2f7cb4e4389/internal/pkg/service/stream/api/service/sink.go#L37
    - https://github.com/keboola/keboola-as-code/blob/ea25882687e215d2e91c352d4aa9b2f7cb4e4389/internal/pkg/service/stream/api/service/sink.go#L104

------------------------

The `Create` endpoint creates the entity in the API call, there it is validated:
https://github.com/keboola/keboola-as-code/blob/ea25882687e215d2e91c352d4aa9b2f7cb4e4389/internal/pkg/service/stream/api/service/sink.go#L25-L29

------------------

The `Update` endpoint needs the old valute from database + there can be a conflict/retry, so the entity is modified in the task/update func:
https://github.com/keboola/keboola-as-code/blob/ea25882687e215d2e91c352d4aa9b2f7cb4e4389/internal/pkg/service/stream/api/service/sink.go#L98-L119

---------------------

It would be better if the Jsonnet validation (even duplicated) was also called within the API call.

It would be easiest to call `updateFn` in the API call as well, but only check the error. The same for Source.

The entity is not stored, so it will not trigger the creation of Storage API resources and other slow things.

Reply:
Please, update also source update, so it will be same.